File: Retrofit excercise/app/src/main/java/hr/ferit/davidbilic/retrofitexcercise/NetworkUtils.java
package hr.ferit.davidbilic.retrofitexcercise;

import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;


public class NetworkUtils {
    private static final String BASE_API="https://makeup-api.herokuapp.com/";
    private static APIInterface apiInterface;

    public static APIInterface getApiInterface() {
        if (apiInterface == null) {
            Retrofit retrofit = new Retrofit.Builder()
                    .baseUrl(BASE_API)
                    .addConverterFactory(GsonConverterFactory.create())
                    .build();

        apiInterface = retrofit.create(APIInterface.class);
        }
        return apiInterface;
    }
}

File: KotlinExcercise-TvShows/app/src/main/java/gg/bidavid/tvshowsapp/networking/NetworkUtils.kt
package gg.bidavid.tvshowsapp.networking

import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory

const val BASE_URL = "https://api.tvmaze.com/"

object NetworkUtils {

    val apiInterface:APIinterface = Retrofit.Builder()
        .baseUrl(BASE_URL)
        .addConverterFactory(GsonConverterFactory.create())
        .build()
        .create(APIinterface::class.java)

}